### ISSUE TITLE
Handle unused variable with `glue`

### ISSUE THREAD

**Issue by denrou** [jimhester/lintr#352]
The following code gives a warning when `lint` is applied:

```r
library(glue)
test <- function() {
  x <- 1
  glue("{x}")
}
```
The warning is `local variable 'x' assigned but may not be used`.

It appears the linter `object_usage_linter` refers to `codetools::checkUsage`, which is not available on github unfortunately (so I can't open an issue). I was wondering if this issue could be handle by `lintr` directly or if a workaround exists.

For now, I applied a `#nolint (unused variable)` for each problematic line.

**Comment by russHyde** [comment]
Reprex:

```
expect_lint("library(glue)\ntest <- function() {\n  x <- 1\nglue::glue('{x}')\n}\n", NULL, object_usage_linter)
```

**Comment by russHyde** [comment]
Similar to the `object_usage_linter` issue in https://github.com/jimhester/lintr/issues/358

**Comment by jimhester** [comment]
The object usage linter now respects variables declared in `utils::globalVariables()`, so you can add them there to avoid these warnings.

**Comment by dragosmg** [comment]
I'm not quite sure why, but declaring global variables with `utils` doesn't seem to work for me.  `object_usage_linter()` still complains when i define a variable and the use it only with `glue::glue`.

I think the way the check has been implemented in `object_usage_linter()` reads the global variables (`declared_globals`) after the package has been installed, while I usually run `lintr::lint_package()` before installing the new version. Is this the desired behaviour or am I missing anything?

**Comment by jimhester** [comment]
You need to install the package before linting it.
